src/form/shared.jl

```diff
@@ -3,22 +3,52 @@
 #################################
 #
 # This is the home of functions that are shared across multiple branches
-# of the type hierarchy.  Hence all function in this file should be over 
+# of the type hierarchy.  Hence all function in this file should be over
 # union types.
 #
-# The types defined in this file should not be exported because they exist 
+# The types defined in this file should not be exported because they exist
 # only to prevent code replication
 #
-# Note that Union types are discouraged in Julia, 
+# Note that Union types are discouraged in Julia,
 # https://docs.julialang.org/en/release-0.6/manual/style-guide/#Avoid-strange-type-Unions-1
 # and should be used with discretion.
 #
 # If you are about to add a union type,
 # first double check if a different type hierarchy can resolve the issue
 # instead.
 #
+AbstractBIMForms = Union{AbstractACPForm, AbstractACTForm, AbstractDCPForm, AbstractWRForm, AbstractWRMForm}
```

Comment:
Note that type unions are discouraged in Julia.  So for now, I would recommend removing this collection and simply say that the `opf_bf` problem does not yet support these forms and defining new forms just for BFMs.

Later on after this PR, I will consider unification of both forms.

Reply:
ok, unification code removed